Added tests for Get-Module over remoting

SteveL-MSFT opened the pull request:
Some additional tests that are remoting specific based on code coverage

Part of https://github.com/PowerShell/PowerShell/issues/4161

SteveL-MSFT committed:
[feature]
added tests for Get-Module over remoting
updated HelpersRemoting.psm1 to support CimSession

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1:
```diff
@@ -0,0 +1,93 @@
+        }
+    }
+
+    It "ArugmentCompleter for PSEdition should work for '<cmd>'" -TestCases @(
```
I see currently we use `TabExpansion2 ` only in two files. Should we put completion tests in one place or put tab completion tests in cmdlet test files?
Make sense make one meta test to cover all (cmdlets) tab completions? It seems something like we do for aliases.

SteveL-MSFT replied:
I'll move this to TabCompletion.Tests.ps1

SteveL-MSFT commented on test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1:
```diff
@@ -0,0 +1,93 @@
+    ) {
+        param($parameter, $value)
+        $parameters = @{$parameter=$value}
+        { Get-Module @parameters } | ShouldBeErrorId "RemoteDiscoveryWorksOnlyInListAvailableMode,Microsoft.PowerShell.Commands.GetModuleCommand"
```
Will add

SteveL-MSFT committed:
[feature]
address PR feedback

daxian-dbw commented:
@SteveL-MSFT can you please add `[feature]` tag and trigger the feature test run? The full build triggered by the first commit failed.

SteveL-MSFT commented on test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1:
```diff
@@ -0,0 +1,80 @@
+        }
+    }
+
+    AfterAll {
```
We've been using this pattern everywhere.  If there is a case where AfterAll/AfterEach doesn't get run, we should open an issue for Pester

SteveL-MSFT commented:
@daxian-dbw done

daxian-dbw commented:
@SteveL-MSFT there is one failure in the new tests. Please fix the test.
```
Describing Remote module tests
VERBOSE: Using Global AppVeyor Credential
VERBOSE: Using Global AppVeyor Credential
 [-] Error occurred in Describe block 425ms
   Method not found: 'Void System.Security.SecureStringMarshal.ZeroFreeCoTaskMemUnicode(IntPtr)'.
   At C:\projects\powershell\test\tools\Modules\HelpersRemoting\HelpersRemoting.psm1:66 char:20
```
Please also include `[Feature]` tag when submitting new commits.

SteveL-MSFT commented:
@daxian-dbw that api doesn't exist in our sources.  I suspect it's in MI.dll.

daxian-dbw commented:
@SteveL-MSFT the error seems to be caused by `$cimsession = New-RemoteSession -CimSession`. It's thrown from `HelpersRemoting.psm1`. It possibly reveals a code bug.

SteveL-MSFT commented:
I'm pretty sure it's a product issue.  However, it's likely an issue with mi.dll.

daxian-dbw commented:
Can you please open an issue with a simple repro step? The tests that require `$CimSession` need to be marked pending.

SteveL-MSFT committed:
[feature]
commented out CimSession use pending https://github.com/PowerShell/PowerShell/issues/4819

SteveL-MSFT commented:
Created https://github.com/PowerShell/PowerShell/issues/4819

daxian-dbw commented:
There are 2 test failurs in AppVeryor test run. Can you please fix them?
